app/src/main/java/com/sopt/dive/ui/components/CommonInputField.kt

```diff
@@ -0,0 +1,63 @@
+package com.sopt.dive.ui.components
+
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+@Composable
+fun CommonInputField(
+    titleText: String,
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeMessage: String,
+    keyboardOptions: KeyboardOptions,
+    visualTransformation: VisualTransformation? = null,
+    errorMessage: String? = null
+
+    ) {
+
+    Column() {
+
+        Text(titleText.uppercase().toString(), modifier = Modifier.fillMaxWidth())
+
+        TextField(
+            value = value,
+            onValueChange = { onValueChange(it) },
```

Comment:
람다는 불필요한 리컴포지션을 발생시킬수 있습니다!
지금 여기서 onValueChange가 (String) -> Unit 콜백이니 람다 없이 전달해줘도 괜찮습니다!
아래처럼 파라미터로 onValueChange를 만들어주시는건 어떠실까요??

```suggestion
            onValueChange = onValueChange,
```

Reply:
오 `onValueChange = onValueChange` 이렇게 작성하면 기존 함수를 그대로 넘겨주면서 해당 함수를 참조하게 되는 거군요! 제가 기존에 작성한 코드는 람다를 새로 정의해서 `it`을 인자로 받아 그 안에서 함수를 호출하는 형태였네요.
두 방식의 차이를 확실히 이해할 수 있어서 도움이 되었습니다. 좋은 정보 감사합니다 😊